azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/WebAppUtils.java

```diff
@@ -93,33 +94,57 @@ public static AppServicePlan createOrGetAppServicePlan(String servicePlanName,
-                    = azure.resourceGroups().contain(servicePlanResGrp) ?
-                    withGroup.withExistingResourceGroup(servicePlanResGrp) :
-                    withGroup.withNewResourceGroup(servicePlanResGrp);
-
-            plan = withPricingTier.withPricingTier(pricingTier).withOperatingSystem(os).create();
-
-            log.info(SERVICE_PLAN_CREATED);
+            plan = createAppServicePlan(servicePlanName, resourceGroup, azure, servicePlanResourceGroup,
+                    region, pricingTier, log, os);
         } else {
-            log.info(String.format(SERVICE_PLAN_EXIST, plan.name(), plan.resourceGroupName()));
+            plan = updateAppServicePlan(plan, pricingTier, log);
         }
-
         return plan;
     }
 
+    public static AppServicePlan createAppServicePlan(String servicePlanName,
+                                                           final String resourceGroup,
+                                                           final Azure azure,
+                                                           final String servicePlanResourceGroup,
+                                                           final Region region,
+                                                           final PricingTier pricingTier,
+                                                           final Log log,
+                                                           final OperatingSystem os) throws MojoExecutionException {
+        if (region == null) {
+            throw new MojoExecutionException("Please config the <region> in pom.xml, " +
+                    "it is required to create a new Azure App Service Plan.");
+        }
+        servicePlanName = AppServiceUtils.getAppServicePlanName(servicePlanName);
+        final String servicePlanResGrp = AppServiceUtils.getAppServicePlanResourceGroup(
+                resourceGroup, servicePlanResourceGroup);
+        log.info(String.format(CREATE_SERVICE_PLAN, servicePlanName));
+
+        final AppServicePlan.DefinitionStages.WithGroup withGroup = azure.appServices().appServicePlans()
+                .define(servicePlanName).withRegion(region);
+
+        final AppServicePlan.DefinitionStages.WithPricingTier withPricingTier
+                = azure.resourceGroups().contain(servicePlanResGrp) ?
+                withGroup.withExistingResourceGroup(servicePlanResGrp) :
+                withGroup.withNewResourceGroup(servicePlanResGrp);
+
+        final AppServicePlan result = withPricingTier.withPricingTier(pricingTier).withOperatingSystem(os).create();
+
+        log.info(SERVICE_PLAN_CREATED);
+        return result;
+    }
+
+    public static AppServicePlan updateAppServicePlan(final AppServicePlan appServicePlan,
+                                                      final PricingTier pricingTier,
+                                                      final Log log) throws MojoExecutionException {
+        log.info(String.format(UPDATE_APP_SERVICE_PLAN));
```

Comment:
null check for appServicePlan and pricingTier

Reply:
Done.